### ISSUE TITLE
man page says timeouts aren't configurable but then describes how to configure them

### ISSUE THREAD

**Issue by xaxxon** [distcc/distcc#212]
https://github.com/distcc/distcc/blob/master/man/distcc.1#L647

**Comment by xaxxon** [comment]
also, what is the difference between DISTCC_FALLBACK and DISTCC_SKIP_LOCAL_RETRY ?  They sound the same.

**Comment by afbjorklund** [comment]
The "dcc_connect_timeout" is not configurable (4 seconds), the "dcc_io_timeout" is (default: 5 minutes)
Main difference is that the I/O timeout also includes the actual compilation, which is why it is configurable*.

* 6ef4e4177bf19a0a78c1fe3e21538696b4ff720c

DISTCC_FALLBACK determines if compile should be done locally instead, if they _fail_ to be distributed.
Since either the network transport or something else failed, we don't know the actual compiler status.

DISTCC_SKIP_LOCAL_RETRY is to avoid recompiling things that _did_ distribute, but failed to compile.
That is: normally there is some problem with the file itself, but this retry is done to detect "other" errors.

**Comment by xaxxon** [comment]
> The timeouts are not configurable at present.

Doesn't this statement refer to both timeouts?   If not, which timeouts (plural) is it referring to?  You mentioned one that isn't configurable, but not a second one.

**Comment by afbjorklund** [comment]
This one is not configurable:

``` C
src/clinet.c:const int dcc_connect_timeout = 4; /* seconds */
```

But this one is, since 6ef4e41:
``` C
src/io.c:    static const int default_dcc_io_timeout = 300; /* seconds */
src/io.c:        current_timeout = default_dcc_io_timeout;
```

**Comment by xaxxon** [comment]
My original intent was to point out that the documentation is wrong and I
think we've established that pretty clearly at this point right?

**Comment by afbjorklund** [comment]
@xaxxon: Yeah, I think the documentation is even more out of the sync (than the code, which is too...)

But if you have some suggestions for improved wording, you should make them into a Pull Request ?

**Comment by craftyguy** [comment]
Any particular reason why this timeout is not configurable:

> src/clinet.c:const int dcc_connect_timeout = 4; /* seconds */

**Comment by afbjorklund** [comment]
Sounds like a DISTCC_CONNECT_TIMEOUT would only be natural

But I guess only @sourcefrog knows why it was removed earlier ?

```
+2004-07-29 19:12:49 GMT        Martin Pool <mbp@sourcefrog.net>        patch-93
+
+    Summary:
+      alarm-based connection timeout

+     * mbp@sourcefrog.net--happy/distcc--devel--2--patch-4
+       remove docs for deleted DISTCC_CONNECT_TIMEOUT

```
